README.md

```diff
@@ -24,14 +24,46 @@ To create the database for development:
+If you want to run the test suite, it's important that the `queue` user be able to create and destroy databases. Keep that in mind if you decide to change the privileges of the `queue` user.
+
+## Database Migrations
+
+We use [Sequelize](http://docs.sequelizejs.com/) as an ORM to make interacting with our database easier. However, its build-in support for migrations is primitive and hard to automate, so Queue includes its own migration support.
+
+Our philosophy is that the Sequelize model definitions in `models/` should serve as the source of truth for what the database should look like at any given repository state. However, because Sequelize can't automatically mutate the database schema to bring it up-to-date with what's expected, we track each change to the database schema so that we can smoothly update the database whenever the schema changes.
+
+Migrations live in the `migrations/` directory and are named like `###_MY_MIGRATION_NAME.sql`. Each one represents a set of database schema changes, which can include adding or removing columns or tables. They are explicitly ordered and uniquely identified by the number at the beginning of their name. A migration should take the existing schema and transform it in such as way that it matches the schema that would be generated by running Sequelize's `sync({ force: true })`. This is enforced by our test suite and CI: if a PR would result in the migration chain being out of sync with the Sequelize models, the build will fail.
+
+### How to modify the database schema
+
+To modify a model definition, you only need to do two things:
+
+- Update the appropriate model definitions in `models/`. Remember that these serve as the source of truth for the database schema.
+
+- Write a migration in `migrations/`. It should use the next available index number and exactly mirror the changes made in the Sequelize model definition. If you're unsure of what this query would look like, you can run the test suite and look at the failure message for the database tests. This will contain one or more queries that should bring the database into the correct state.
+
+Once a migration has been merged into `master`, it should be treated as immutable. Modifying an old migration could produce unintended consequences if future migrations depend on it.
+
+### Technical details of migrations
+
+Migrations are tracked in a `migrations` table, which stores the index of the migration, its filename, and when it was applied.
+
+The following steps occur when the app starts up in order to perform migrations (these are implemented in `migrations/index.js`):
+
+1. The database is queried to find the index of the last migration that was applied.
+
+2. All `.sql` files from `migrations` are read, parsed, and sorted by index.
+
+3. Each one is applied to the database sequentially in order. When a migration is successfully applied, an entry is created in the `migrations` table so that that migration won't be applied again.
```

Comment:
sequentially == in order, no? :)

Reply:
True, I guess "in series" would be the better term. Was trying to emphasize that they happen one after the other 😛